service/src/test/java/bio/terra/workspace/service/resource/controlled/cloud/azure/AzureStorageAccessServiceUnitTest.java

```diff
@@ -422,19 +424,128 @@ void createAzureStorageContainerSasToken_blobName() throws InterruptedException
             List.of(
                 SamConstants.SamControlledResourceActions.READ_ACTION,
                 SamConstants.SamControlledResourceActions.WRITE_ACTION));
+    setupMocks(storageAccountResource, storageContainerResource, Optional.empty());
 
     var result =
         azureStorageAccessService.createAzureStorageContainerSasToken(
-            UUID.randomUUID(),
-            storageContainerResource,
-            storageAccountResource,
-            startTime,
-            expiryTime,
+            storageContainerResource.getWorkspaceId(),
+            storageContainerResource.getResourceId(),
             userRequest,
-            null,
-            "foo/the/bar.baz",
-            null);
+            new SasTokenOptions(null, startTime, expiryTime, "foo/the/bar.baz", null));
 
     assertValidToken(result.sasToken(), BlobContainerSasPermission.parse("racwdl"), true);
   }
+
+  @Test
+  public void createAzureStorageContainerSasToken_basedOnLzSharedStorageAccount_Success()
+      throws InterruptedException {
+    var storageContainerResource =
+        buildStorageContainerResource(
+            PrivateResourceState.NOT_APPLICABLE,
+            AccessScopeType.ACCESS_SCOPE_SHARED,
+            ManagedByType.MANAGED_BY_USER,
+            null);
+
+    var ipRange = "168.1.5.60-168.1.5.70";
+    when(mockSamService()
+            .listResourceActions(
+                ArgumentMatchers.eq(userRequest),
+                ArgumentMatchers.eq(storageContainerResource.getCategory().getSamResourceName()),
+                ArgumentMatchers.eq(storageContainerResource.getResourceId().toString())))
+        .thenReturn(
+            List.of(
+                SamConstants.SamControlledResourceActions.WRITE_ACTION,
+                SamConstants.SamControlledResourceActions.READ_ACTION));
+    setupMocks(null, storageContainerResource, Optional.of(true));
+
+    var result =
+        azureStorageAccessService.createAzureStorageContainerSasToken(
+            storageContainerResource.getWorkspaceId(),
+            storageContainerResource.getResourceId(),
+            userRequest,
+            new SasTokenOptions(ipRange, startTime, expiryTime, null, null));
+
+    assertValidToken(result.sasToken(), BlobContainerSasPermission.parse("racwdl"), false);
+    assertTrue(
+        result.sasToken().contains("sip=" + ipRange),
+        "the SignedIP was added to the query parameters");
+  }
+
+  @Test
+  public void createAzureStorageContainerSasToken_LzSharedStorageAccountNotFound_Failure()
+      throws InterruptedException {
+    var storageContainerResource =
+        buildStorageContainerResource(
+            PrivateResourceState.NOT_APPLICABLE,
+            AccessScopeType.ACCESS_SCOPE_SHARED,
+            ManagedByType.MANAGED_BY_USER,
+            null);
+
+    var ipRange = "168.1.5.60-168.1.5.70";
```

Comment:
Why add the ip range in this test?

Reply:
It is just an option for SAS token. It is not specific to this test. We can remove it or use another options.

Reply:
Got it; yeah I'd suggest removing it.